lib/templates.ts

```diff
@@ -324,35 +309,27 @@ export async function getAllTemplates(
 }
 
 /**
- * Get all form templates for a specific User.
+ * Get all form templates for the User calling the function.
  * @returns An array of Form Records
  */
 export async function getAllTemplatesForUser(
   ability: UserAbility,
-  userID: string,
-  isPublished?: boolean
+  options?: {
+    sortByDateUpdated?: "asc" | "desc";
+    requestedWhere?: Prisma.TemplateWhereInput;
+  }
 ): Promise<Array<FormRecord>> {
   try {
     checkPrivileges(ability, [{ action: "view", subject: "FormRecord" }]);
-    const canUserAccessAllTemplates = checkPrivilegesAsBoolean(ability, [
```

Comment:
I'm not quite sure I understand.  The function does not accept a different user identifier.  The logged in user would only be able to retrieve templates that they are an owner on.  This would include someone with Manage All Forms, they would only receive templates that they personally own and not all the system templates.